training-react/tic-tac-toe/src/app/components/App/components/Game/index.js

```diff
@@ -1,70 +1,55 @@
 import style from './styles.scss';
 import Moves from './components/Moves';
 
-class Game extends Component {
-  state = {
-    history: [{ squares: Array(9).fill(null), step: 'step0' }],
-    xIsNext: true,
-    stepNumber: 0
-  };
+const Game = ({ history, stepNumber, xIsNext, handleClick, jumpTo }) => {
+  const current = history[stepNumber];
+  const winner = calculateWinner(current.squares);
 
-  calculateWinner = squares => {
-    for (let i = 0; i < gameConsts.lines.length; i += 1) {
-      const [a, b, c] = gameConsts.lines[i];
-      if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
-        return squares[a];
-      }
-    }
-    return null;
-  };
-
-  handleClick = i => {
-    const history = this.state.history.slice(0, this.state.stepNumber + 1);
-    const current = history[history.length - 1];
-    const squares = current.squares.slice();
-    const step = `step${this.state.stepNumber + 1}`;
-    if (this.calculateWinner(squares) || squares[i]) {
-      return;
-    }
-
-    squares[i] = this.state.xIsNext ? 'X' : 'O';
-
-    this.setState({
-      history: history.concat([{ squares, step }]),
-      stepNumber: history.length,
-      xIsNext: !this.state.xIsNext
-    });
-  };
-
-  jumpTo = step => {
-    this.setState({
-      stepNumber: step,
-      xIsNext: step % 2 === 0
-    });
-  };
-
-  render() {
-    const history = this.state.history;
-    const current = history[this.state.stepNumber];
-    const winner = this.calculateWinner(current.squares);
-
-    const status = winner ? `Winner: ${winner}` : `Next player: ${this.state.xIsNext ? 'X' : 'O'}`;
-    return (
-      <div className={style.game}>
-        <div>
-          <Board squares={current.squares} onClick={this.handleClick} />
-        </div>
-        <div className={style.gameInfo}>
-          <div>{status}</div>
-          {<Moves history={history} jump={this.jumpTo} />}
-        </div>
+  const status = winner ? `Winner: ${winner}` : `Next player: ${xIsNext ? 'X' : 'O'}`;
+  return (
+    <div className={style.game}>
+      <div>
+        <Board squares={current.squares} onClick={handleClick} />
       </div>
-    );
-  }
-}
-
-export default Game;
+      <div className={style.gameInfo}>
+        <div>{status}</div>
+        {<Moves history={history} jump={jumpTo} />}
+      </div>
+    </div>
+  );
+};
+
+const mapStateToProps = state => ({
+  history: state.game.history,
+  xIsNext: state.game.xIsNext,
+  stepNumber: state.game.stepNumber
+});
+
+const mapDispatchToProps = dispatch => ({
+  jumpTo: step => dispatch(actionCreators.jumpToStep(step)),
+  handleClick: i => dispatch(actionCreators.clickSquare(i))
+});
+
+Game.propTypes = {
+  history: arrayOf(
+    shape({
+      squares: arrayOf(string)
+    }).isRequired
+  ),
+  xIsNext: bool.isRequired,
+  stepNumber: number.isRequired,
+  jumpTo: func.isRequired,
+  handleClick: func.isRequired
+};
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(Game);
```

Comment:
Let's split this component in a smart and presentational component. All redux connection should be placed in container component, and int the presentational one u must only handle the layout.
Check [this](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0) and [this](https://medium.com/@learnreact/container-components-c0e67432e005)